magda-gateway/src/setupTenantMode.ts

```diff
@@ -0,0 +1,28 @@
+import TenantsLoader from "./reloadTenants";
+
+export var multiTenantsMode: boolean = undefined;
```

Comment:
I guess you mean `TenantMode` instead of `TenantConfig`.
I think your approach is good in the sense that all three values are grouped in `TenantMode` type. However, a tenant mode value will still need to be exported as mutable variable for module createBaseProxy to use. That is, a variable `tenantMode: TenantMode` needs to be declared in module `buildApp`, assigned value by ` setupTenantMode()` then exported.
Does my reasoning make sense?

Reply:
Done -- now immutable.